src/iff.imageio/iffinput.cpp

```diff
@@ -452,27 +514,30 @@ IffInput::read_native_tile(int subimage, int miplevel, int x, int y, int /*z*/,
     if (!seek_subimage(subimage, miplevel))
         return false;
 
-    if (m_buf.empty())
-        readimg();
+    if (m_buf.empty()) {
+        if (!readimg()) {
+            errorfmt("IFF error could not read image for tile");
```

Comment:
If readimg() itself calls errorfmt for each way that it can fail, you don't need to do it again here.